src/Compilers/Core/Portable/SourceGeneration/GeneratorDriver.cs

```diff
@@ -176,10 +178,29 @@ internal GeneratorDriverState RunGeneratorsCore(Compilation compilation, Diagnos
                     generatorState = ex is null
                                      ? new GeneratorState(context.InfoBuilder.ToImmutable())
                                      : SetGeneratorException(MessageProvider, GeneratorState.Uninitialized, generator, ex, diagnosticsBag, isInit: true);
+
+                    // invoke the post init callback if requested
+                    if (generatorState.Info.PostInitCallback is object)
```

Comment:
Should post init be skipped if `generatorState.Exception` is set?

Reply:
Ah, it is already, because we don't use the returned context when the exception is set (we pass GeneratorState.Uninitialized above)

---
In reply to: [564886706](https://github.com/dotnet/roslyn/pull/50026#discussion_r564886706) [](ancestors = 564886706)